api/jobs.go

```diff
@@ -119,12 +124,64 @@ func (api *JobStoreAPI) GetJobsHandler(w http.ResponseWriter, req *http.Request)
 		http.Error(w, serverErrorMessage, http.StatusInternalServerError)
 		return
 	}
-
 }
 
 // unlockJob unlocks the provided job lockID and logs any error with WARN state
 func (api *JobStoreAPI) unlockJob(ctx context.Context, lockID string) {
+	log.Event(ctx, "Entering unlockJob function, which unlocks the provided job lockID.", log.INFO)
 	if err := api.jobStore.UnlockJob(lockID); err != nil {
 		log.Event(ctx, "error unlocking lockID for a job resource", log.WARN, log.Data{"lockID": lockID})
 	}
 }
+
+// PutNumTasksHandler returns a function that updates the number_of_tasks in an existing Job resource, which is associated with the id passed in.
+func (api *JobStoreAPI) PutNumTasksHandler(ctx context.Context) http.HandlerFunc {
+	log.Event(ctx, "Creating handler function, which calls PutNumberOfTasks to update the number_of_tasks in the job with the supplied id.", log.INFO)
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		id := vars["id"]
+		count := vars["count"]
+		logData := log.Data{"id": id, "count": count}
+
+		numTasks, err := strconv.Atoi(count)
+		if err != nil {
+			log.Event(ctx, "invalid path parameter - count should be a positive integer", log.Error(err), logData, log.ERROR)
+			http.Error(w, "invalid path parameter - count should be a positive integer", http.StatusBadRequest)
```

Comment:
"failed to convert count to integer"

Reply:
I've changed the message as suggested. I had left it as it was because, although the count would not be found negative at this point, it is always the case that the count needs to be a positive integer. However, having said that, I do agree that it looked a little strange and I had previously thought about just saying "count should be an integer" instead.